File: examples/governance_static_go/domain/registered_framework.go
package domain

import "time"

type RegisteredFramework struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	Name string `json:"name"`
	Jurisdiction string `json:"jurisdiction"`
	Version string `json:"version"`
	Authority string `json:"authority"`
	EffectiveDate time.Time `json:"effective_date"`
	Requirements []FrameworkRequirement `json:"requirements"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e RegisteredFramework) EventName() string { return "RegisteredFramework" }

func (e RegisteredFramework) GetOccurredAt() time.Time { return e.OccurredAt }

File: examples/governance/compliance_domain/lib/compliance_domain/regulatory_framework/events/registered_framework.rb
module ComplianceDomain
  class RegulatoryFramework
    module Events
      class RegisteredFramework
        attr_reader :aggregate_id, :name, :jurisdiction, :version, :authority, :effective_date, :requirements, :status, :occurred_at

        def initialize(aggregate_id: nil, name: nil, jurisdiction: nil, version: nil, authority: nil, effective_date: nil, requirements: nil, status: nil)
          @aggregate_id = aggregate_id
          @name = name
          @jurisdiction = jurisdiction
          @version = version
          @authority = authority
          @effective_date = effective_date
          @requirements = requirements
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end
